compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/QueryExpressionNode.java

```diff
@@ -43,11 +43,14 @@ public QueryPipelineNode queryPipeline() {
 
     @Deprecated
     public SelectClauseNode selectClause() {
-        ClauseNode endClause = endClause();
-        return endClause.kind() == SyntaxKind.SELECT_CLAUSE ? (SelectClauseNode) endClause : null;
+        ClauseNode endClause = resultClause();
+        if (endClause.kind() != SyntaxKind.SELECT_CLAUSE) {
+            throw new IllegalStateException("Query expression having collect-clause does not contain select-clause");
```

Comment:
As per the concluded discussion: https://github.com/ballerina-platform/ballerina-lang/pull/40537#discussion_r1209971133
```suggestion
            throw new IllegalStateException("select-clause not found");
```